File: Single LinkedList/Python/ReverseLinkedList.py
class Node:
    def __init__(self, data):
        self.data=data;
        self.next=None;

class LinkedList:
    def __init__(self):
        self.head=None;
        
    def insert_at_end(self, data):
        new_node=Node(data);
        if self.head is None:
            self.head=new_node;
            return;
        last_node=self.head;
        while last_node.next:
            last_node=last_node.next;
        last_node.next=new_node;
    
    def print_list(self):
        current_node=self.head;
        while current_node:
            print(current_node.data, end=" -> ");
            current_node=current_node.next;
        print("None");
        
    def reverse(self):
        if self.head is None or self.head.next is None:
            return;
        temp=None;
        while self.head:
            next=self.head.next;
            self.head.next=temp;
            temp=self.head;
            self.head=next;
        self.head=temp;
        
if __name__=="__main__":
    linked_list=LinkedList();
    for i in range(1, 6):
        linked_list.insert_at_end(i);
        
    print("LinkedList before reversing ");
    linked_list.print_list();
    
    linked_list.reverse();
    print("LinkedList before reversing ");
    linked_list.print_list();

File: Single LinkedList/Java/ReverseLinkedList.java
class Node {
    int data;
    Node next;
    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList{
    Node head;
    public LinkedList() {
        head = null;
    }

    public void insert_at_end(int data) {
        Node new_node = new Node(data);
        if (head == null) {
            head = new_node;
            return;
        }
        Node temp = head;
        while (temp.next != null) {
            temp = temp.next;
        }
        temp.next = new_node;
    }

    public void print_list() {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("null");
    }

    public void reverse(){
        if(head==null || head.next==null) 
        {
            return;
        }
        Node temp=null;
        while(head!=null)
        {
            Node next=head.next;
            head.next=temp;
            temp=head;
            head=next;
        }
        head=temp;
    }
}

public class ReverseLinkedList {
    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        for(int i=1;i<6;i++)
        {
            list.insert_at_end(i);
        }
        System.out.println("LinkedList before reversing ");
        list.print_list();

        list.reverse();
        System.out.println("LinkedList after reversing ");
        list.print_list();
    }
}
